lib/puppet/parser/environment_compiler.rb

```diff
@@ -82,4 +136,12 @@ def evaluate_applications
       end
     end
   end
+
+  # Overrides the regular compiler to be able to return the list of resources after a prune
+  # has taken place in the graph representation. Before a prune, the list is the same as in the regular
+  # compiler
+  #
+  def resources
+    @pruned_resources || super
```

Comment:
it is indeed an override of of the @resource attribute - the alternative would be to poke a new value into @resources (which is really horrible as the super class declares it). We could do this with a protected setter